src/ParseObject.js

```diff
@@ -933,6 +933,27 @@ class ParseObject {
     return false;
   }
 
+  /**
+   * Returns true if this object exists on the Server
+   *
+   * @param {Object} options
+   * Valid options are:<ul>
+   *   <li>useMasterKey: In Cloud Code and Node only, causes the Master Key to
+   *     be used for this request.
+   *   <li>sessionToken: A valid session token, used for making a request on
+   *       behalf of a specific user.
+   * </ul>
+   * @return {Promise<boolean>} A boolean promise that is fulfilled if object exists.
+   */
+  async exists(options?: RequestOptions): Promise<boolean> {
+    try {
+      await this.fetch(options);
+      return true;
+    } catch (e) {
+      return false;
```

Comment:
I think we should check `e.code === 101` and only return false in this case. Otherwise `throw e`. I am afraid that a network issue could be interpreted as an inexistent object.

Reply:
I was thinking here again. Maybe do a query by id would be better here? I am not sure if the developer, when calling an `exists()` function, is expecting its data to be updated.

Reply:
Thats a good point.